https://bugzilla.mozilla.org/show_bug.cgi?id=1975253

with forkserver enabled, setproctitle butchers argv, which would prevent
content processes to find their actual binary. cache binaryPath before
setproctitle is called to work around it.

https://phabricator.services.mozilla.com/D258832

Index: ipc/glue/ForkServer.cpp
--- ipc/glue/ForkServer.cpp.orig
+++ ipc/glue/ForkServer.cpp
@@ -30,6 +30,11 @@
 #  include "mozilla/SandboxLaunch.h"
 #endif
 
+#if defined(XP_OPENBSD)
+#  include "BinaryPath.h"
+#  include <err.h>
+#endif
+
 #include <algorithm>
 
 namespace mozilla {
@@ -74,6 +79,13 @@ ForkServer::ForkServer(int* aArgc, char*** aArgv) : mA
  */
 static void ForkServerPreload(int& aArgc, char** aArgv) {
   Omnijar::ChildProcessInit(aArgc, aArgv);
+#if defined(XP_OPENBSD)
+  char binaryPath[MAXPATHLEN];
+  nsresult rv = mozilla::BinaryPath::Get(binaryPath);
+  if (NS_FAILED(rv)) {
+    errx(1, "failed to cache binary path ?");
+  }
+#endif
 }
 
 /**
